api/v3/MembershipType.php

```diff
@@ -47,6 +47,10 @@ function civicrm_api3_membership_type_create($params) {
       $params[$field] = implode(CRM_Core_DAO::VALUE_SEPARATOR, $params[$field]);
     }
   }
+
+  if (!isset($params['domain_id'])) {
+    $params['domain_id'] = CRM_Core_Config::domainID();
```

Comment:
This would incorrectly be set if 'id' is set - I think we should spin this out & resolve it first - but I think we should support 'current_domain' in the same way we support 'user_contact_id', now - ie. ie would look like

$params['domain_id']['api.default'] => 'current_domain_id';

I would need to understand from @totten why the code in utils handles '('or similar') but I guess we could just 'special-handle' the specific string 'current_domain_id'


  if (!empty($fieldValue) || $fieldValue === '0' || $fieldValue === 0) {
    // if value = 'user_contact_id' (or similar), replace value with contact id
    if (!is_numeric($fieldValue) && is_scalar($fieldValue)) {
      $realContactId = _civicrm_api3_resolve_contactID($fieldValue);
      if ('unknown-user' === $realContactId) {
        throw new API_Exception("\"$fieldName\" \"{$fieldValue}\" cannot be resolved to a contact ID", 2002, array('error_field' => $fieldName, "type" => "integer"));
      }